### ISSUE TITLE
iMac with Radeon 580, less than 4MH/s, why?

### ISSUE THREAD

**Issue by tinue** [Genoil/cpp-ethereum#260]
This project so far seems to be the only one actually working on MacOS, thanks for this! What I am asking is not an issue with this project: I could reproduce the same thing using Bootcamp under Windows 10. Nevertheless, I don't know where else to ask ;)

Anyway: This is what gets reported with `./ethminer -G --list-devices`:
`[OPENCL]:
Listing OpenCL devices.
FORMAT: [deviceID] deviceName
[0] AMD Radeon Pro 580 Compute Engine
	CL_DEVICE_TYPE: GPU
	CL_DEVICE_GLOBAL_MEM_SIZE: 8589934592
	CL_DEVICE_MAX_MEM_ALLOC_SIZE: 2147483648
	CL_DEVICE_MAX_WORK_GROUP_SIZE: 256`

And this is the benchmark result:
`min/mean/max: 3754894/3826289/3844778 H/s`

I expected something around 24MH/s for a Radeon 580, and not less than 4.

Does anyone know why the iMac performs that poorly?

**Comment by EoD** [comment]
You could have a look at ethereum-mining/ethminer#90 or ethereum-mining/ethminer#44, the issues might be related?

**Comment by mxjoe** [comment]
I have the same issue with a MacBook Pro (Late 2016). The miner always uses the integrated and slow Intel graphics, although I tell it to use the Radeon Pro 460. This seems to be a problem which at least several people have.

**Comment by EoD** [comment]
@mxjoe Did you try the newer miner at ethereum-mining/ethminer instead of this one?

**Comment by mxjoe** [comment]
@EoD Yes, It tried that one too, and it didn't work either. BUT I then found [wtfuzz's fork](https://github.com/wtfuzz/ethminer) of ethminer and with this it works and I get significantly higher bitrate.

**Comment by tinue** [comment]
Sorry, I didn't see your comment (missed the e-mail alert). Thanks for the pointer! It appears that the patch would roughly double the mining rate. In my case, I would need a five fold increase to match the hash rate of a regular Windows PC with a plugged-in Radeon 580. 

For whatever reason, the built-in Radeon 580 in an iMac is 5 times slower (possibly 2.5 times slower with the patch) compared to a PCI connected Radeon 580. As I said, this is also true when I use the Windows 10 version of the miner on the same iMac.

It could still be a software issue, of course, which is Mac (and not Operating System) specific.

The miner does show the Radeon card (see my original issue posting), and I launch it with "-G", so it should pick the Radeon chipset. How can I tell if still uses the Intel Integrated Graphics instead of the Radeon?

**Comment by EoD** [comment]
@mxjoe wtfuzz's changes are in ethereum-mining/ethminer#44 which I linked above. You probably don't want to use wtfuzz's fork directly, otherwise you are missing out quite some bug fixes/improvements.

**Comment by mxjoe** [comment]
@tinue On a 2017 iMac with Radeon Pro 580 I get approx. 19,5 MH/s with Claymore Dual Miner.

@EoD Thanks for your advice, I thought that too but I need to find out first how I can use issue #44 improvements with current ethereum-mining/ethminer as the fix in issue #44 didn#t make it into ethminer's master yet.
